src/lang/modifyAst/boolean.spec.ts

```diff
@@ -17,26 +46,32 @@ describe('boolean', () => {
       solidIds: number[],
       toolIds: number[],
       instance: ModuleType,
+      kclManager: KclManager,
       rustContext: RustContext
     ) {
-      const ast = assertParse(code, instance)
-      if (err(ast)) throw ast
-
-      const { artifactGraph } = await enginelessExecutor(
-        ast,
-        undefined,
-        undefined,
-        rustContext
-      )
-      const sweeps = [...artifactGraph.values()].filter(
-        (n) => n.type === 'sweep'
+      const { artifactGraph, ast } = await getAstAndArtifactGraph(
+        code,
+        instance,
+        kclManager
       )
+      // path selection is what p&c is doing with the filter (v. sweep)
+      const paths = [...artifactGraph.values()].filter((n) => n.type === 'path')
       const solids: Selections = {
-        graphSelections: solidIds.map((i) => sweeps[i]),
+        graphSelections: solidIds.map((i) => {
+          return {
+            artifact: paths[i],
+            codeRef: paths[i].codeRef,
+          }
+        }),
```

Comment:
This is what was preventing me from seeing the fix in action. `solidIds.map((i) => sweeps[i])` was wrong, leading to graphSelections[*].artifact to be undefined, and that would skip the last-child-lookup. And preventing me from seeing the issue